### PR TITLE
Task: create a repo. Executor: Oleksii Bashkirov

### PR DESCRIPTION
**Pull request by OleksiiBashkirov**
Created the Book entity, repository and service;
Also created the OnlineBookStoreApplication class with the main method;
Added application.properties file for database configuration.
Configured Checkstyle and Continuous Integration (CI) using GitHub Actions.
Added H2 database configuration for testing purposes.

### CODE REVIEW

File: src/main/java/mate/academy/model/Book.java [hunk 1]
```diff
@@ -20,23 +20,19 @@
 @AllArgsConstructor
 @Builder
 public class Book {
+    //comment
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/model/Book.java]: remove 

```
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
You must merge you changes to master only when mentor approve them

**Comment by OleksiiBashkirov** [comment]
> You must merge you changes to master only when mentor approve them

Yes, thanks for the guidance. Initially, I committed and pushed changes directly to the main branch. After realizing that I should have used a different branch, I corrected the workflow by creating the book-entity branch for the changes. Moving forward, I will ensure to wait for your approval before merging any changes to the main branch. Thank you for your understanding and support.

**Comment by OleksiiBashkirov** [comment]
I have reset the "main" branch to ensure it only contains approved commits. All changes are now correctly placed in the "book-entity" branch, and I will await your approval before merging any changes to the "main" branch moving forward.
Thank you for your guidance.

File: .mvn/wrapper/maven-wrapper.properties [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```
Comment (MAMentorFX) [inline .mvn/wrapper/maven-wrapper.properties:1]: I believe this also should be in gitignore

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -0,0 +1,8 @@
+spring.datasource.url=jdbc:mysql://localhost:3306/book_repository?serverTimeZone=UTC
+spring.datasource.username=root
+spring.datasource.password=Root1234
+spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
+
+spring.jpa.hibernate.ddl-auto=create-drop
+spring.jpa.show-sql=true
+#comment
```
Comment (MAMentorFX) [inline src/main/resources/application.properties]:
```suggestion
```

**Review by MAMentorFX — APPROVED** [review]

File: .mvn/wrapper/maven-wrapper.properties (hunk 2 above)
Comment (OleksiiBashkirov) [reply .mvn/wrapper/maven-wrapper.properties:1]: Hi, thanks. I have updated the ".gitignore" file to exclude the ".mvn" directory and removed the ".mvn" directory from the repository as per your suggestion.
Thank you for your guidance.

**Pull request merged**